internal/runner/command_test.go

```diff
@@ -0,0 +1,132 @@
+package runner
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"syscall"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/kballard/go-shellquote"
+)
+
+func TestRunAndForwardSignal(t *testing.T) {
+	cmd := exec.Command("echo", "hello world")
+
+	err := runAndForwardSignal(cmd)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestRunAndForwardSignal_CommandExitsWithNonZero(t *testing.T) {
+	cmd := exec.Command("false")
+
+	err := runAndForwardSignal(cmd)
+	exitError := new(exec.ExitError)
+	if errors.As(err, &exitError) && exitError.ExitCode() != 1 {
+		t.Errorf("Expected exit code 1, but got %d", exitError.ExitCode())
+	}
+}
+
+func TestRunAndForwardSignal_SignalReceivedInMainProcess(t *testing.T) {
+	cmd := exec.Command("sleep", "10")
+
+	// Send a SIGTERM signal to the main process.
+	go func() {
+		pid := os.Getpid()
+		process, err := os.FindProcess(pid)
+		if err != nil {
+			fmt.Println(err)
```

Comment:
```suggestion
			t.Errorf("os.FindProcess(%d) error = %v", err)
```
Four things here:

1. If `os.FindProcess` fails to find the PID of the current process, it's a bad time for the computer. So if this happens, the test is probably invalid and should be recorded as a failure.
2. Stdout (used by `fmt.Println`) is not preferable when it could be logged as part of the test output instead.
3. We can't use `t.Fatalf` because that calls `t.FailNow` which per documentation shouldn't be used outside of the main test goroutine.
4. Logging bare errors isn't good practice - aim to include some context in the log message. Since this is a test, that usually means the name and arguments of the function call returning the error.